faucet/src/main.rs

```diff
@@ -19,14 +24,38 @@ struct Opts {
     /// Keyring / keyfile options.
     #[clap(flatten)]
     account_info: runtime::cli::ProviderUserOpts,
-
+    
     /// Connection settings for the BTC Parachain.
     #[clap(flatten)]
     parachain: runtime::cli::ConnectionOpts,
-
+    
     /// Settings specific to the faucet client.
     #[clap(flatten)]
     faucet: FaucetConfig,
+    
+    /// Allowance config
+    #[clap(long, default_value = "./faucet-allowance-config.json")]
+    allowance_config: PathBuf,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct AllowanceAmount {
+    ticker: String,
+    amount: u128,
+}
+
+type AllowanceTuple = (CurrencyId, u128);
+type Allowance = Vec<AllowanceTuple>;
+impl From<&AllowanceAmount> for AllowanceTuple {
+    fn from(allowance_amount: &AllowanceAmount) -> Self {
+        (CurrencyId::try_from_symbol(allowance_amount.ticker.clone()).expect("Failed to parse input ticker"), allowance_amount.amount)
```

Comment:
We should only convert into the `CurrencyId` type before submitting an extrinsic (see for example [here](https://github.com/interlay/interbtc-clients/blob/d5af9c500e1739675bd4d960828cecb5fa6f5fd6/oracle/src/main.rs#L122-L123) in the oracle client) this is because we lazily load these at runtime (which requires a parachain connection first): https://github.com/interlay/interbtc-clients/blob/d5af9c500e1739675bd4d960828cecb5fa6f5fd6/runtime/src/rpc.rs#L574

Reply:
Actually I just noticed this is called immediately after establishing a parachain connection so it might be fine, but still I would only convert before submission to make sure we use the most up-to-date view.